pkg/client/client.go

```diff
@@ -205,6 +208,63 @@ func NewClientFromKeys(accessKey, secretAccessKey, region string) (Client, error
 	}, nil
 }
 
+type describeRegionsData struct {
+	err                   error
+	describeRegionsOutput *ec2.DescribeRegionsOutput
+	lastUpdated           time.Time
+}
+
+// Do not access directly, use cachedAWSDescribeRegions() instead.
+var regionsCache describeRegionsData
+
+// cachedDescribeRegions returns full list of regions from AWS.
+// DescribeRegionsOutput is cached to avoid AWS API calls on each reconcile loop.
+func cachedAWSDescribeRegions(awsSession *session.Session) (*ec2.DescribeRegionsOutput, error) {
+	if regionsCache.describeRegionsOutput != nil && regionsCache.err == nil &&
```

Comment:
I think the region cache needs to have some way of being keyed against the account ID from the session, else you might give data from one account to another if users were running machines across multiple accounts